application/src/main/java/com/github/pieter_groenendijk/repository/AccountRepository.java

```diff
@@ -0,0 +1,89 @@
+package com.github.pieter_groenendijk.repository;
+
+import java.util.Date;
+import java.util.UUID;
+import com.github.pieter_groenendijk.model.Account;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.cfg.Configuration;
+import java.util.List;
+import com.github.pieter_groenendijk.hibernate.SessionFactoryFactory;
+
+public class AccountRepository implements IAccountRepository {
+
+    private SessionFactory sessionFactory;
+
+    public AccountRepository() {
+        sessionFactory = new SessionFactoryFactory().create();
+        System.out.println("Repo1");
+    }
+
+    public Account retrieveById(long id) {
+        Session session = sessionFactory.openSession();
+
+        Account account = null;
+        try {
+            session.beginTransaction();
+            long accountId = id;
+            account = session.get(Account.class, accountId);
+
+            if (account != null) {
+                System.out.println("Account found: " + account.getEmail());
+            } else {
+                System.out.println("No account found with ID: " + accountId);
+            }
+            session.getTransaction().commit();
+        } catch (Exception e) {
+            if (session.getTransaction() != null) {
+                session.getTransaction().rollback();
+            }
+            System.out.println("Data retrieval failed!");
+            e.printStackTrace();
+        } finally {
+            session.close();
+        }
+        return account;
```

Comment:
Dit returned mogelijk null. Dat is mogelijk klote voor de functies die deze functie callen.

Reply:
Hier is nu een aanpassing voor gedaan. Als je nu een ID uitvraagt die niet bekend is, krijg je een 404 terug met een keurige responsebody.